Target "otel_ngx_module" links to target "CURL::libcurl" but the target was   not found.

Hi, 

I am working on a universal docker-build container image to build the NGINX OTEL module. 

My baseimage is `ubunut:20.04`.  After building grpc, cmake, googletest with benchmark and opentelemetry I was able to start building the NGINX Module.

While running `cmake` I received the following error:

```
root@825df8ee05d0:/opentelemetry-cpp-contrib/instrumentation/nginx/build# cmake ..
-- nginx: using version 1.19.9
-- nginx: --with-compat enabled
-- Configuring done
CMake Error at CMakeLists.txt:12 (add_library):
  Target "otel_ngx_module" links to target "CURL::libcurl" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```
I have modified the `CMakeLists.txt`
and have added 

```
find_package( CURL REQUIRED )
```
With this small fix its working just fine. Is this a docker issue? Have you seen this issue before?

**Steps to reproduce**
`git clone` inside the docker container and issue
`cmake ..`

**What is the expected behavior?**
No errors just my lovely NGINX Module file 💚

**What is the actual behavior?**
Error message shown above

--- a/instrumentation/nginx/CMakeLists.txt
+++ b/instrumentation/nginx/CMakeLists.txt
@@ -6,6 +6,7 @@ find_package(opentelemetry-cpp REQUIRED)
 find_package(Threads REQUIRED)
 find_package(protobuf REQUIRED)
 find_package(gRPC REQUIRED)
+find_package(CURL REQUIRED)
 
 include(${CMAKE_CURRENT_SOURCE_DIR}/nginx.cmake)
 
@@ -42,4 +43,5 @@ target_link_libraries(otel_ngx_module
   PRIVATE
   ${OPENTELEMETRY_CPP_LIBRARIES}
   gRPC::grpc++
+  CURL::libcurl
 )

--- a/instrumentation/nginx/src/otel_ngx_module.cpp
+++ b/instrumentation/nginx/src/otel_ngx_module.cpp
@@ -1,4 +1,3 @@
-#include <opentelemetry/nostd/mpark/variant.h>
 #include <opentelemetry/sdk/trace/processor.h>
 #include <opentelemetry/trace/span.h>
 #include <unordered_map>

--- a/instrumentation/nginx/src/propagate.cpp
+++ b/instrumentation/nginx/src/propagate.cpp
@@ -2,7 +2,6 @@
 #include "location_config.h"
 #include "nginx_utils.h"
 #include <opentelemetry/context/context_value.h>
-#include <opentelemetry/nostd/mpark/variant.h>
 #include <opentelemetry/trace/propagation/b3_propagator.h>
 #include <opentelemetry/trace/propagation/http_trace_context.h>
 #include <opentelemetry/trace/span.h>

--- a/instrumentation/nginx/test/instrumentation/lib/mix/tasks/dockerfiles.ex
+++ b/instrumentation/nginx/test/instrumentation/lib/mix/tasks/dockerfiles.ex
@@ -2,7 +2,7 @@ defmodule Mix.Tasks.Dockerfiles do
   use Mix.Task
 
   @grpc_version "v1.36.4"
-  @otel_cpp_version "v0.7.0"
+  @otel_cpp_version "v1.0.0-rc1"
 
   def run([out_dir | combos]) do
     out_dir_abs = Path.expand(out_dir)

